BlueprintUICommonControls/Tests/Sources/ImageTests.swift

```diff
@@ -46,4 +37,355 @@ class ImageTests: XCTestCase {
         compareSnapshot(of: element, size: CGSize(width: 100, height: 100))
     }
 
+    func test_measure_aspectFill() {
+        func validate(
+            size: CGSize,
+            constraint: SizeConstraint,
+            expectedValue: CGSize,
+            line: UInt = #line
+        ) {
+            self.validate(
+                contentMode: .aspectFill,
+                imageSize: size,
+                constraint: constraint,
+                expectedValue: expectedValue,
+                line: line
+            )
+        }
+
+        // Wide aspect ratio image
+
+        validate(
+            size: .init(width: 20, height: 10),
+            constraint: .init(width: .unconstrained, height: .unconstrained),
+            expectedValue: .init(width: 20, height: 10)
+        )
+
+        validate(
+            size: .init(width: 20, height: 10),
+            constraint: .init(width: .atMost(40), height: .unconstrained),
+            expectedValue: .init(width: 40, height: 20)
+        )
+
+        validate(
+            size: .init(width: 20, height: 10),
+            constraint: .init(width: .atMost(10), height: .unconstrained),
+            expectedValue: .init(width: 10, height: 5)
+        )
+
+        validate(
+            size: .init(width: 20, height: 10),
+            constraint: .init(width: .atMost(5), height: .atMost(5)),
+            expectedValue: .init(width: 10, height: 5)
+        )
+
+        validate(
+            size: .init(width: 20, height: 10),
+            constraint: .init(width: .atMost(40), height: .atMost(40)),
```

Comment:
I'd like to see a case where the constraint is fixed but is has a "longer" aspect ratio than the image, e.g. 40 x 10. This only exercises one side of the `CGSize(width: width, height: height).aspectRatio > imageSize.aspectRatio` comparison.

Same request for the tall image, and for both wide & tall in `aspectFit`

Reply:
47adc98